File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/typegroup/fetchoptions/TypeGroupAssignmentFetchOptions.java
/*
 *  Copyright ETH 2025 Zürich, Scientific IT Services
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package ch.ethz.sis.openbis.generic.asapi.v3.typegroup.fetchoptions;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.fetchoptions.PersonFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.typegroup.TypeGroupAssignment;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;

@JsonObject("as.dto.typegroup.fetchoptions.TypeGroupAssignmentFetchOptions")
public class TypeGroupAssignmentFetchOptions extends FetchOptions<TypeGroupAssignment> implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private PersonFetchOptions registrator;

    @JsonProperty
    private TypeGroupAssignmentSortOptions sort;

    public PersonFetchOptions withRegistrator()
    {
        if (registrator == null)
        {
            registrator = new PersonFetchOptions();
        }
        return registrator;
    }

    public PersonFetchOptions withRegistratorUsing(PersonFetchOptions fetchOptions)
    {
        return registrator = fetchOptions;
    }

    public boolean hasRegistrator()
    {
        return registrator != null;
    }

    @Override
    public TypeGroupAssignmentSortOptions sortBy()
    {
        if (sort == null)
        {
            sort = new TypeGroupAssignmentSortOptions();
        }
        return sort;
    }

    @Override
    public TypeGroupAssignmentSortOptions getSortBy()
    {
        return sort;
    }

    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("TypeGroupAssignment", this);
        f.addFetchOption("Registrator", registrator);
        return f;
    }
}

File: api-openbis-javascript/src/v3/as/dto/typegroup/fetchoptions/TypeGroupAssignmentFetchOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/person/fetchoptions/PersonFetchOptions",
		"as/dto/typegroup/fetchoptions/TypeGroupFetchOptions", "as/dto/sample/fetchoptions/SampleTypeFetchOptions",
		"as/dto/typegroup/fetchoptions/TypeGroupAssignmentSortOptions" ], function(require, stjs, FetchOptions) {
	var TypeGroupAssignmentFetchOptions = function() {
	};
	stjs.extend(TypeGroupAssignmentFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.typegroup.fetchoptions.TypeGroupAssignmentFetchOptions';
		constructor.serialVersionUID = 1;
		prototype.registrator = null;
		prototype.typeGroup = null;
		prototype.sampleType = null;
		prototype.sort = null;

		prototype.withRegistrator = function() {
			if (this.registrator == null) {
				var PersonFetchOptions = require("as/dto/person/fetchoptions/PersonFetchOptions");
				this.registrator = new PersonFetchOptions();
			}
			return this.registrator;
		};
		prototype.withRegistratorUsing = function(fetchOptions) {
			return this.registrator = fetchOptions;
		};
		prototype.hasRegistrator = function() {
			return this.registrator != null;
		};

		prototype.withTypeGroup = function() {
			if (this.typeGroup == null) {
				var TypeGroupFetchOptions = require("as/dto/typegroup/fetchoptions/TypeGroupFetchOptions");
				this.typeGroup = new TypeGroupFetchOptions();
			}
			return this.typeGroup;
		};
		prototype.withTypeGroupUsing = function(fetchOptions) {
			return this.typeGroup = fetchOptions;
		};
		prototype.hasTypeGroup = function() {
			return this.typeGroup != null;
		};

		prototype.withSampleType = function() {
			if (this.sampleType == null) {
				var SampleTypeFetchOptions = require("as/dto/sample/fetchoptions/SampleTypeFetchOptions");
				this.sampleType = new SampleTypeFetchOptions();
			}
			return this.sampleType;
		};
		prototype.withSampleTypeUsing = function(fetchOptions) {
			return this.sampleType = fetchOptions;
		};
		prototype.hasSampleType = function() {
			return this.sampleType != null;
		};

		prototype.sortBy = function() {
			if (this.sort == null) {
				var TypeGroupAssignmentSortOptions = require("as/dto/typegroup/fetchoptions/TypeGroupAssignmentSortOptions");
				this.sort = new TypeGroupAssignmentSortOptions();
			}
			return this.sort;
		};
		prototype.getSortBy = function() {
			return this.sort;
		};
	}, {
		registrator : "PersonFetchOptions",
		typeGroup : "TypeGroupFetchOptions",
		sampleType : "SampleTypeFetchOptions",
		sort : "TypeGroupAssignmentSortOptions"
	});
	return TypeGroupAssignmentFetchOptions;
})